include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -162,6 +169,239 @@ class alignment_file_format_sam
+              header_type                                             & header_ptr,
+              seq_type                                                & seq,
+              qual_type                                               & qual,
+              id_type                                                 & id,
+              offset_type                                             & offset,
+              ref_seq_type                                            & SEQAN3_DOXYGEN_ONLY(ref_seq),
+              ref_id_type                                             & ref_id,
+              ref_offset_type                                         & ref_offset,
+              align_type                                              & align,
+              flag_type                                               & flag,
+              mapq_type                                               & mapq,
+              mate_type                                               & mate,
+              tag_dict_type                                           & tag_dict,
+              e_value_type                                            & SEQAN3_DOXYGEN_ONLY(e_value),
+              bit_score_type                                          & SEQAN3_DOXYGEN_ONLY(bit_score))
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        auto next = view::take_until_or_throw(is_char<'\t'>);
+
+        // these variables need to be stored to compute the ALIGNMENT
+        std::conditional_t<!detail::decays_to_ignore_v<ref_offset_type>, ref_offset_type, size_t> ref_offset_tmp{};
+        std::conditional_t<!detail::decays_to_ignore_v<offset_type>, offset_type, size_t> offset_tmp{};
+        [[maybe_unused]] size_t soft_clipping_end{};
+        [[maybe_unused]] std::vector<std::pair<char, size_t>> cigar{};
+        [[maybe_unused]] size_t ref_length{0}, seq_length{0}; // length of aligned part for ref and query
+
+        // Fields 1-5: ID FLAG REF_ID REF_OFFSET MAPQ
+        // -------------------------------------------------------------------------------------------------------------
+        read_field(stream_view | next, id);
+        ++ranges::begin(stream_view); // skip tab
+
+        read_field(stream_view | next, flag);
+        ++ranges::begin(stream_view);
+
+        read_field(stream_view | next, ref_id);
+        ++ranges::begin(stream_view);
+
+        read_field(stream_view | next, ref_offset_tmp);
+        ++ranges::begin(stream_view);
+        --ref_offset_tmp; // SAM format is 1-based but SeqAn operates 0-based.
+        if constexpr (!detail::decays_to_ignore_v<ref_offset_type>)
+            ref_offset = ref_offset_tmp;
```

Comment:
you don't need the `if constexpr` check because even if ref_offset is std::ignore, assigning to it will work. By definition you can assign anything to std::ignore.